File: backend/app/schemas/resume.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ResumeOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    candidate_id: uuid.UUID
    original_file_name: str
    file_type: str
    version: int
    is_active: bool
    parsing_status: str
    parsing_method: str | None
    parsing_error: str | None
    created_at: datetime


class ResumeParsedDataOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    parsing_status: str
    parsing_method: str | None
    parsed_data: dict | None

File: src/types/resume.ts
// Matches the backend ResumeOut (schemas/resume.py). Parsing runs asynchronously
// on the server, so a freshly uploaded resume comes back as "pending".
export interface Resume {
    id: string;
    candidate_id: string;
    original_file_name: string;
    file_type: string;
    version: number;
    is_active: boolean;
    parsing_status: "pending" | "processing" | "completed" | "failed";
    parsing_method?: string | null;
    parsing_error?: string | null;
    created_at: string;
}

// Kept as an alias since the upload response has the same shape as ResumeOut.
export type UploadResumeResponse = Resume;
